accepted/00000-upgrading-bootstrap.md

```diff
@@ -0,0 +1,70 @@
+
+# Motivation
+[motivation]: #motivation
+
+There are two main reasons to upgrade Bootstrap: ease of maintenance, and security issues.
+
+Bootstrap 3 comes from an era where browser support was more complicated and it reflects in the stylesheets we have. The current bundled stylesheet for Suma is roughly 10k lines of CSS, which is hard to reason about and maintain to say the least. This problem isn't solely due to Bootstrap, but it is the primary source of the issue. The manner in which Suma has been styled over the years has been essentially add only: since the styles are tightly coupled yet globally shared, removing things has been hard while adding a single override for any given problem is fairly easy. This has resulted in our own themes becoming complex as well, on top of an already large Bootstrap base.  
+
+As a separate issue, Bootstrap 3 has known security vulnerabilities, such as XSS injection, for which there won't be a patch. These security issues by themselves do not pose too serious of a threat due to the nature of the product, but they might be leveraged to create a chain of attacks. Given Suma is open source and has many moving parts, this is a very realistic attack vector.  
+
+The core proposition of this RFC can be summarized in two steps:  
+
+ 1. Simplify our existing stylesheets to reduce the use of custom overrides, and try and compose views and layouts using standard Bootstrap solutions. This step is beneficial even without the Bootstrap upgrade in mind, as currently our stylesheets are simply very convoluted and hard to reason about.  
+ 2. Upgrade Bootstrap. This includes many moving parts which will be outlined below.  
+
+Coming out successfully from the other side, we would hope to have simpler stylesheets with less to maintain, and fewer security issues.  
+
+
+# Detailed design
+[design]: #detailed-design
+
+## Theme & styles
+
+Bootstrap 3 uses Less while Bootstrap 5 only offers Sass styles. This switch by itself is not very problematic, however there are additional issues that make the upgrade more difficult. Our current styles make heave use of numerous mixins from Bootstrap 3 which no longer exist in Bootstrap 5. Additionally, there are numerous breaking styling changes going from 3 to 4 and even more so moving from 4 to 5.  
+
+Instead of trying to manually migrate our existing very large theme source, I propose we instead try to first reduce our existing theme footprint as much as possible. We already have component scoped styles working and in use, so we can try move component or view specific styles (e.g. table styles, setup wizard etc) into component scoped styles and remove them from the global stylesheet.  
```

Comment:
What I would like to point out here in reducing the footprint itself is making sure to reduce also the layers that make the footprint what it is like. It's not only about the width of it but also (and more complicated) about the depth. What I've found the most frustrating part to maintain was the multiple overriding pieces they are just partially overlapping again and again through the layers.
It is by design that the history made it complicated: overriding spacewalk with a full copy of SUSE Manager and/or Uyuni. But this is not the only complexity aspect: the core theme itself is made of overrides and fixes on top to adjust the behavior as desired.
So carving out specific components would help a lot reducing the **width**, but I'd really suggest to squash the **depth** of  components, and I expect this last point to be the most complicated one actually.
For instance, for a table component style, at the moment we have:
```
base theme style +
base table style +
Uyuni theme style fully overriding base theme style +
Uyuni table style fully overriding the base table style
```
and if you apply a single change in the middle of this chain, the result can be different.

Ideally we should have one and only one `theme style` (maybe more a _neutral-branded_ one), then single components would run over it and that's it.

@Etheryte : technically speaking, from your point of view, what would be a good approach to squash the duplicated theme styles into only one?